bcda/auth/api_test.go

```diff
@@ -45,56 +47,106 @@ func (s *AuthAPITestSuite) SetupSuite() {
 
 func (s *AuthAPITestSuite) SetupTest() {
 	s.rr = httptest.NewRecorder()
+	s.server = httptest.NewServer(s.CreateRouter())
 }
 
-func (s *AuthAPITestSuite) TestAuthToken() {
-	clientID, clientSecret, accessToken := uuid.New(), uuid.New(), uuid.New()
-	mock := &auth.MockProvider{}
-	mock.On("MakeAccessToken", auth.Credentials{ClientID: clientID, ClientSecret: clientSecret}).
-		Return(accessToken, constants.ExpiresInDefault, nil)
-	mock.On("MakeAccessToken", auth.Credentials{ClientID: "not_a_client", ClientSecret: "not_a_secret"}).
-		Return("", "", errors.New("some auth error"))
-	auth.SetMockProvider(s.T(), mock)
-
-	// Missing authorization header
-	req := httptest.NewRequest("POST", constants.TokenPath, nil)
-	handler := http.HandlerFunc(auth.GetAuthToken)
-	handler.ServeHTTP(s.rr, req)
-	assert.Equal(s.T(), http.StatusBadRequest, s.rr.Code)
+func (s *AuthAPITestSuite) TestGetAuthTokenErrorSwitchCases() {
+	const errorHappened = "Error Happened!"
+	const errMsg = "Error Message"
 
-	// Malformed authorization header
-	s.rr = httptest.NewRecorder()
-	req = httptest.NewRequest("POST", constants.TokenPath, nil)
-	req.Header.Add("Authorization", "Basic not_an_encoded_client_and_secret")
+	req, err := http.NewRequest("POST", fmt.Sprintf("%s/auth/token", s.server.URL), nil)
+	if err != nil {
+		assert.FailNow(s.T(), err.Error())
+	}
+	//req.Header.Add("Authorization", fmt.Sprintf("Basic %s", tt.authHeader))
 	req.Header.Add("Accept", constants.JsonContentType)
-	handler = http.HandlerFunc(auth.GetAuthToken)
-	handler.ServeHTTP(s.rr, req)
-	assert.Equal(s.T(), http.StatusBadRequest, s.rr.Code)
+	req.SetBasicAuth("good", "client")
+
+	client := s.server.Client()
+
+	tests := []struct {
+		ScenarioName          string
+		ErrorToReturn         error
+		StatusCode            int
+		HeaderRetryAfterValue string
+	}{
+		{"Token Request Timeout Error Return 503", &customErrors.RequestTimeoutError{Err: errors.New(errorHappened), Msg: errMsg}, 503, "1"},
+		{"Token Unexpected SSAS Error Return 500", &customErrors.UnexpectedSSASError{Err: errors.New(errorHappened), Msg: errMsg}, 500, constants.EmptyString},
```

Comment:
add one more scenario for InternalParsingError is returned during MakeAccessToken, and verify 500 is the status code